vm/src/obj/objfloat.rs

```diff
@@ -558,6 +600,20 @@ impl TryFromObject for IntoPyFloat {
     }
 }
 
+pub enum ComplexFloatValue {
```

Comment:
This is good trial for mixed return type. But in this case, I think this function is a simple combination of 2 functions:
- float ** positive float -> float
- float ** negative float -> complex

By keeping float_pow as it was and adding a new function (probably `float_negative_pow`?), it would be well-defined without this type.

`might_complex_op` can be a pow function which call 2 pow funcitons by v2. Roughly, it would look like:
```rust
fn _pow(v1: float, v2: float) -> PyResult {
   if v2 is positive {
       float_pow().into_pyobject()
   } else {
       float_negative_pow().into_pyobject()
   }
}
```

Reply:
Thank you for your kind review :)
I updated PR with reference to your suggestion. Please take a look ;)
If you have any more advice, it would be welcomed!